### PR TITLE
Fillet polyline - maximum radius factor  

### PR DESCRIPTION
**Pull request by vkter**
## maximum radius factor  
Added a maximum radius factor input.
Fixed error when radius is 0.
Fixed error when fillet Tangent

### CODE REVIEW

**Comment by Durman** [comment]
I'm not sure I understand the purpose of that extra parameter. It seems redundant to me. The factor can be controlled by radius.

image

**Comment by vkter** [comment]
It will assign a reasonable maximum radius range to each chamfer
image

**Comment by Durman** [comment]
I see, probably it has sense to have the feature in this way. Did you test the clipping in cyclic mode?

**Comment by vkter** [comment]
This is in cyclic mode
image

**Comment by Durman** [comment]
It seems clipping does not work properly with this setup.

[untitled1.zip](https://github.com/nortikin/sverchok/files/10000237/untitled1.zip)

image

**Comment by vkter** [comment]
The maximum radius limit depends on the length of the shortest adjacent edge. This simple method seems unable to cope with complex situations，It only works well on non concave polygons. Maybe I should find a better way to determine this value

**Comment by vkter** [comment]
I changed the code last night, it's working properly now
image

**Comment by vkter** [comment]
在连接模式和相邻圆角刚好接触时会引发错误，这似乎是曲线连接模块的问题，它无法连接这样的圆角，这并非我添加了bug
image

**Comment by portnov** [comment]
Doesn't this require an Mk2 of the node?

**Comment by Durman** [comment]
> Doesn't this require an Mk2 of the node?

I guess yes.

@vkter http://nortikin.github.io/sverchok/docs/contributing/node_api.html#upgrade-node

Alternative way is to add an option to make new (clamping) behavior switched off by default.

**Comment by vkter** [comment]
It seems better to add a clamping option that is turned off by default. I will continue to modify the code

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "commentTranslate.source": "Bing"
```
Comment (Durman) [inline .vscode/settings.json:2]: I'm not sure we need this in Git. I guess `.vscode` should be added to the git ignore file. Everything else seems okay.

Comment (vkter) [reply .vscode/settings.json:2]: Yes, just ignore it，

**Review by Durman — APPROVED** [review]

**Comment by Durman** [comment]
Congratulations on the first contribution =)

**Pull request merged**